src/mesh/index_derivs.cxx

```diff
@@ -677,11 +681,11 @@ const Field2D Mesh::applyXdiff(const Field2D &var, Mesh::deriv_func func,
           s.p = var[i.xp()];
           s.pp = var[i.xpp()];
 
-          if ((location == CELL_CENTRE) && (loc == CELL_XLOW)) {
+          if (outloc == CELL_XLOW) {
             // Producing a stencil centred around a lower X value
             s.pp = s.p;
             s.p = s.c;
-          } else if (location == CELL_XLOW) {
+          } else if (inloc == CELL_XLOW) {
```

Comment:
Is there another possibility at this point? If not, this could lose the `if` statement.

Reply:
:+1: that one can go